kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/k8s/KubeClusterResource.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.systemtests.k8s;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.kroxylicious.systemtests.k8s.cluster.KubeCluster;
+import io.kroxylicious.systemtests.k8s.cmd.KubeCmdClient;
+
+/**
+ * A Junit resource which discovers the running cluster and provides an appropriate KubeClient for it,
+ * for use with {@code @BeforeAll} (or {@code BeforeEach}.
+ * For example:
+ * <pre><code>
+ *     public static KubeClusterResource testCluster = new KubeClusterResources();
+ *
+ *     &#64;BeforeEach
+ *     void before() {
+ *         testCluster.before();
+ *     }
+ * </code></pre>
+ */
+public class KubeClusterResource {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(KubeClusterResource.class);
+
+    private KubeCluster kubeCluster;
+    private KubeCmdClient cmdClient;
+    private KubeClient client;
+    private static KubeClusterResource kubeClusterResource;
+
+    private String namespace;
+    private String testNamespace;
+
+    /**
+     * Gets instance.
+     *
+     * @return the instance
+     */
+    public static synchronized KubeClusterResource getInstance() {
```

Comment:
If this class is a singleton, it should have a private ctor.

Reply:
done